hack/releasing/promote_pull.sh

```diff
@@ -68,8 +69,9 @@ declare -r REBASE_MAGIC=".git/rebase-apply"
 DRY_RUN=${DRY_RUN:-""}
 KUBERNETES_K8S_IO_UPSTREAM_REMOTE=${KUBERNETES_K8S_IO_UPSTREAM_REMOTE:-upstream}
 KUBERNETES_K8S_IO_FORK_REMOTE=${KUBERNETES_K8S_IO_FORK_REMOTE:-origin}
-KUBERNETES_K8S_IO_MAIN_REPO_ORG=$(git remote get-url "$KUBERNETES_K8S_IO_UPSTREAM_REMOTE" | cut -d: -f2- | cut -d/ -f1)
-KUBERNETES_K8S_IO_MAIN_REPO_NAME=$(git remote get-url "$KUBERNETES_K8S_IO_UPSTREAM_REMOTE" | cut -d: -f2- | cut -d/ -f2- | sed 's/\.git$//')
+KUBERNETES_K8S_IO_MAIN_REPO_ORG=${KUBERNETES_K8S_IO_MAIN_REPO_ORG:-$(git remote get-url "${KUBERNETES_K8S_IO_UPSTREAM_REMOTE}" | awk '{gsub(/http[s]:\/\/|git@/,"")}1' | awk -F'[@:./]' 'NR==1{print $3}')}
+KUBERNETES_K8S_IO_MAIN_REPO_NAME=${KUBERNETES_K8S_IO_MAIN_REPO_NAME:-$(git remote get-url "${KUBERNETES_K8S_IO_UPSTREAM_REMOTE}" | awk '{gsub(/http[s]:\/\/|git@/,"")}1' | awk -F'[@:./]' 'NR==1{print $4}')}
```

Comment:
The main issue was here: we were parsing only SSH URLs, but we also need to handle https:// URLs.

That's why we had:

```
+++ Creating a pull request on GitHub at mimowo:kueue-promote-0.15 for main
expected the "[HOST/]OWNER/REPO" format, got "[//github.com/kubernetes/k8s.io](https://kubernetes.slack.com//github.com/kubernetes/k8s.io)"
```